### PR TITLE
Add model specific patches

### PR DESCRIPTION
**Pull request by YaoJiayi**
Please resolve https://github.com/LMCache/lmcache-vllm/pull/22 first.

Fix errors for llama-8b, mistral-7b and chatglm-9b

### CODE REVIEW

File: lmcache_vllm/vllm_adapter.py [hunk 1]
```diff
@@ -85,6 +95,27 @@ def init_lmcache_engine(
 
     return engine
 
+def broadcast_seq_group_metadata(
```
Comment (ApostaC) [inline lmcache_vllm/vllm_adapter.py]: I guess this function is already in PR #22?

File: lmcache_vllm/vllm_adapter.py [hunk 2]
```diff
@@ -347,13 +388,31 @@ def lmcache_retrieve_kv(
     slot_mapping = model_input.attn_metadata.slot_mapping.flatten()
     seq_lens = model_input.attn_metadata.seq_lens
 
-    if hasattr(model_executable.model, "start_layer"):
-        start_layer = model_executable.model.start_layer
+    
+    if model_name in SUPPORTED_MODELS.llama_family or \
```
Comment (ApostaC) [inline lmcache_vllm/vllm_adapter.py]: Can we have a separate function for this? I'm thinking about this: 
- the function takes in the model_executable and the model_name as input
- the function returns all the metadata we need in a data class object

**Review by ApostaC — COMMENTED** [review]
Thanks, Jiayi! Please see the comments below

File: lmcache_vllm/vllm_adapter.py (hunk 2 above)
Comment (YaoJiayi) [reply lmcache_vllm/vllm_adapter.py]: No problem.

File: lmcache_vllm/vllm_adapter.py (hunk 1 above)
Comment (YaoJiayi) [reply lmcache_vllm/vllm_adapter.py]: Yeah. I branched this from #22 . Should be gone after merging the #22

File: lmcache_vllm/vllm_adapter.py [hunk 3]
```diff
@@ -509,17 +570,20 @@ def lmcache_retrieve_kv(
...
                 ops.reshape_and_cache_flash(
                     kv_tuple[layer_idx][0].to(key_cache.device),
                     kv_tuple[layer_idx][1].to(value_cache.device),
                     key_cache,
                     value_cache,
                     slot_mapping[start_pos:start_pos + lmc_num_computed_tokens],
-                    layer.self_attn.attn.kv_cache_dtype,
-                    layer.self_attn.attn._k_scale,
-                    layer.self_attn.attn._v_scale,
+                    attn_layer.attn.kv_cache_dtype,
```
Comment (ApostaC) [inline lmcache_vllm/vllm_adapter.py:584]: Does the attn_layer in all the models has the fields `attn.kv_cache_dtype`, `attn._k_scale`, and `attn._v_scale`? 
If not, how about we put `kv_cache_dtype`, `k_scale` and `v_scale` in `ModelInputSubset` and extract them based on the model name in `create_model_input_subset`?

**Review by ApostaC — COMMENTED** [review]
Please see the comment below

File: lmcache_vllm/vllm_adapter.py (hunk 3 above)
Comment (YaoJiayi) [reply lmcache_vllm/vllm_adapter.py:584]: I checked the vllm source code. All models should have these fields (https://github.com/vllm-project/vllm/blob/main/vllm/attention/layer.py#L59).

**Review by ApostaC — APPROVED** [review]
LGTM!

**Pull request merged**